### PR TITLE
Replace Mongo tests with updated tests.

### PR DESCRIPTION
**Pull request by tylerbenson**
I think these were the last remaining "legacy" integration tests!

Done in preparation for changes to Mongo instrumentation.

### CODE REVIEW

File: dd-java-agent/instrumentation/mongo/driver-async-3.3/src/test/groovy/MongoAsyncClientTest.groovy [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import org.bson.Document
+import spock.lang.Shared
+import spock.lang.Timeout
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CountDownLatch
+
+import static datadog.trace.agent.test.utils.TraceUtils.runUnderTrace
+
+@Timeout(10)
```
Comment (mar-kolya) [inline dd-java-agent/instrumentation/mongo/driver-async-3.3/src/test/groovy/MongoAsyncClientTest.groovy]: Is this still needed?

File: dd-java-agent/instrumentation/mongo/driver-3.1/driver-3.1.gradle [hunk 2]
```diff
@@ -28,9 +28,8 @@ dependencies {
   implementation deps.autoservice
 
   testCompile project(':dd-java-agent:testing')
-  
-  testCompile group: 'org.assertj', name: 'assertj-core', version: '2.9.+'
-  testCompile group: 'org.mockito', name: 'mockito-core', version: '2.19.0'
+  testCompile project(':dd-java-agent:instrumentation:mongo').sourceSets.test.output
+  testCompile group: 'de.flapdoodle.embed', name: 'de.flapdoodle.embed.mongo', version: '1.50.5'
```
Comment (mar-kolya) [inline dd-java-agent/instrumentation/mongo/driver-3.1/driver-3.1.gradle]: Did you consider using mongo container instead? just and idea

Comment (tylerbenson) [reply dd-java-agent/instrumentation/mongo/driver-3.1/driver-3.1.gradle]: I've heard other tracers have had bad luck with it.  This seemed to be pretty reliable, so I didn't want to fuss with it.

File: dd-java-agent/instrumentation/mongo/driver-async-3.3/src/test/groovy/MongoAsyncClientTest.groovy (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/mongo/driver-async-3.3/src/test/groovy/MongoAsyncClientTest.groovy]: I like adding timeouts when making blocking calls and it was cleaner to add it once here than on each `get()` below.

**Pull request merged**